fontbe/src/features.rs

```diff
@@ -72,12 +76,153 @@ impl Display for NotSupportedError {
+                .variable_axes
+                .iter()
+                .enumerate()
+                .map(|(i, a)| {
+                    (
+                        a.tag,
+                        (
+                            &a.converter,
+                            fea_rs::compile::AxisInfo {
+                                index: i as u16,
+                                default_value: a.default.into(),
+                                min_value: a.min.into(),
+                                max_value: a.max.into(),
+                            },
+                        ),
+                    )
+                })
+                .collect(),
+            axes: static_metadata
+                .variable_axes
+                .iter()
+                .map(|a| (a.tag, a))
+                .collect(),
+            static_metadata,
+        }
+    }
+}
+
+impl<'a> VariationInfo for FeaVariationInfo<'a> {
+    fn axis_info(&self, axis_tag: font_types::Tag) -> Option<fea_rs::compile::AxisInfo> {
+        self.fea_rs_axes.get(&axis_tag).map(|a| a.1)
+    }
+
+    fn normalize_coordinate(
+        &self,
+        axis_tag: font_types::Tag,
+        value: font_types::Fixed,
+    ) -> font_types::F2Dot14 {
+        let converter = &self
+            .axes
+            .get(&axis_tag)
+            .expect("Unsupported axis")
+            .converter;
+        let user_coord = UserCoord::new(value.to_f64() as f32);
+        F2Dot14::from_f32(user_coord.to_normalized(converter).to_f32())
+    }
+
+    fn resolve_variable_metric(
+        &self,
+        values: &HashMap<BTreeMap<Tag, Fixed>, i16>,
+    ) -> Result<
+        (
+            i16,
+            Vec<(write_fonts::tables::variations::VariationRegion, i16)>,
+        ),
+        Box<(dyn StdError + 'static)>,
+    > {
+        // WARNING: this will fail if the fea location isn't also a glyph location. In time we may wish to fix that.
+        let var_model = &self.static_metadata.variation_model;
+
+        // Compute deltas using f64 as 1d point and delta, then ship them home as i16
+        let point_seqs: HashMap<_, _> = values
+            .iter()
+            .map(|(pos, value)| {
+                let user = UserLocation::from_iter(
+                    pos.iter()
+                        .map(|(tag, value)| (*tag, UserCoord::new(value.to_f64() as f32))),
+                );
+                (user.to_normalized(&self.axes), vec![*value as f64])
+            })
+            .collect();
+
+        // Only 1 value per region for our input
+        // TODO is that actually guaranteed?
+        let deltas: Vec<_> = var_model
+            .deltas(&point_seqs)?
+            .into_iter()
+            .map(|(region, values)| {
+                assert!(values.len() == 1, "{} values?!", values.len());
+                (region, values[0])
+            })
+            .collect();
+
+        // Compute the default on the unrounded deltas
+        let default_value = deltas
```

Comment:
> // Compute the default

hm I'd expect the "default" to _not_ be computed, but already defined in the input (as we discussed in https://github.com/googlefonts/fontc/pull/345)